spec/support/fake_analytics.rb

```diff
@@ -172,11 +175,41 @@ def browser_attributes
 end
 
 RSpec.configure do |c|
+  groups = []
+
   c.around do |ex|
     keys = Array(ex.metadata[:allowed_extra_analytics])
     FakeAnalytics::UndocumentedParamsChecker.allowed_extra_analytics = keys
     ex.run
+
+    if keys.present?
+      group = ex.example_group
+      group = group.superclass until [nil, RSpec::Core::ExampleGroup].include?(group.superclass)
+      groups << [group, FakeAnalytics::UndocumentedParamsChecker.checked_extra_analytics.to_a]
+    end
   ensure
     FakeAnalytics::UndocumentedParamsChecker.allowed_extra_analytics = []
+    FakeAnalytics::UndocumentedParamsChecker.checked_extra_analytics = []
+  end
+
+  c.after(:all) do |ex|
+    groups.group_by(&:first).each do |group, pairs|
+      allowed_extra_analytics = group.metadata[:allowed_extra_analytics]
+      next if allowed_extra_analytics.blank?
+      all_checked_extra_analytics = pairs.map(&:last).flatten.uniq
+      if allowed_extra_analytics.include?(:*)
+        expect(all_checked_extra_analytics).not_to(
+          be_blank,
```

Comment:
positive assertions sometimes easier to follow?
```suggestion
        expect(all_checked_extra_analytics).to(
          be_present,
```

Reply:
Yep, good call 👍  Updated in 6ef50b6.